tests/unit/rule_strings/pipeline.txt

```diff
@@ -0,0 +1,58 @@
+import linker.utilities.validation_utils as validation_utils
+rule all:
+    input:
+        final_output=['{snake_dir}/result.parquet'],
+        validation='{snake_dir}/input_validations/final_validator'
+rule:
+    name: "results_validator"
+    input: ['{snake_dir}/result.parquet']
+    output: touch("{snake_dir}/input_validations/final_validator")
+    localrule: True         
+    message: "Validating results input"
+    run:
+        for f in input:
+            validation_utils.validate_dummy_file(f)
+rule:
+    name: "step_1_python_pandas_validator"
+    input: ['{test_dir}/input_data1/file1.csv', '{test_dir}/input_data2/file2.csv']
+    output: touch("{snake_dir}/input_validations/step_1_python_pandas_validator")
+    localrule: True         
```

Comment:
Remind me what this does? Did you confirm it behaves as expected?

Reply:
the `touch` creates an empty file at the `filename` listed if the validation passes
`localrule` means that it will run locally even if other rules are run on the cluster; we don't have slurm support back yet so it's hard to tell whether it does that or not.
In general, I recall we took care to consider whether the valdiation properly blocks the running of the next stage of the pipeline--this method definitely does so.